bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoVacuumHandler.java

```diff
@@ -549,14 +551,12 @@ private void updateHistoryRecordLegacy(JsonArray historyData) {
     private void updateHistoryRecord(HistoryRecordDTO historyRecordDTO) {
         JsonObject historyRecord = GSON.toJsonTree(historyRecordDTO).getAsJsonObject();
         if (historyRecordDTO.getStart() != null) {
-            historyRecord.addProperty("start", historyRecordDTO.getStart().split("\\+")[0].split("\\-")[0]);
-            updateState(CHANNEL_HISTORY_START_TIME,
-                    new DateTimeType(historyRecordDTO.getStart().split("\\+")[0].split("\\-")[0]));
+            historyRecord.addProperty("start", historyRecordDTO.getStart().substring(0, 19));
```

Comment:
The previous implementation was more dynamic, now the length is hardcoded, it might be usefull to check the length and write to debug if not meets the expected length, does this make sense?

Expand the condition in line 5553 if `historyRecordDTO.getStart().length > 19`  ?  
(same for getEnd())

Reply:
yes, makes sense. I'll take a look indeed, I have some idea to solve it.

Reply:
Think this is resolved now